### PR TITLE
Add TEX parsing integration tests

### PR DESCRIPTION
**Pull request by dorianvp**
This PR adds an integration test which tries to parse 2 TEX addresses, a correct one and a random string.

### CODE REVIEW

**Comment by dorianvp** [comment]
Now that I think about it, maybe we should move the **wrong** parsing to another separate function, and create functions for parsing on each network. What do you think @juanky201271 ?

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt [hunk 1]
```diff
@@ -447,3 +447,41 @@ class ExecuteSaplingBalanceFromSeed {
...
+
+        val setCrytoProvider = uniffi.zingo.setCryptoDefaultProviderToRing()
+        println(setCrytoProvider)
+
+        val initFromSeedJson: String = uniffi.zingo.initFromSeed(server, seed, birthday, datadir, chainhint, monitorMempool)
+        println("\nInit from seed:")
+        println(initFromSeedJson)
+        val initFromSeed: InitFromSeed = mapper.readValue(initFromSeedJson)
+        assertThat(initFromSeed.seed).isEqualTo(Seeds.HOSPITAL)
+        assertThat(initFromSeed.birthday).isEqualTo(1)
```
Comment (juanky201271) [inline android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt]: For clarity here I would use the vars: seed & birthday...

File: android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt [hunk 2]
```diff
@@ -447,3 +447,41 @@ class ExecuteSaplingBalanceFromSeed {
...
+
+        val result: String = uniffi.zingo.executeCommand("parse_address", "texregtest1z754rp9kk9vdewx4wm7pstvm0u2rwlgy4zp82v")
+        println("\nParsed Address:")
+        println(result)
+
+        assertThat(result).isNotEmpty()
+        assertThat(result).isNotNull()
```
Comment (juanky201271) [inline android/app/src/androidTest/java/org/ZingoLabs/Zingo/RustFFITest.kt:489]: Take a look at this:
https://github.com/zingolabs/zingo-mobile/blob/0888420b39079f0fe5ce3617fef1f092d5c8eae6/app/utils/Utils.ts#L310
The result of this command is a bit weird... you can have a `null` string, or an string beginning by `Error`, or a valid JSON object with the field `success` with value `success` and chain_name... this validation is a bit complex...

**Review by juanky201271 — CHANGES_REQUESTED** [review]

**Comment by juanky201271** [comment]
> Now that I think about it, maybe we should move the **wrong** parsing to another separate function, and create functions for parsing on each network. What do you think @juanky201271 ?

IMO to have 4 separate tests would be the best option. I think we have to use `regtest` as much as we can and don't rely in any external server. And, please put those 4 tests in CI if you can. Thanks, good job.

BTW I need to test it locally ASAP.

File: rust/zingomobile_utils/src/lib.rs [hunk 3]
```diff
@@ -25,25 +30,55 @@ pub fn android_integration_test(abi: &str, test_name: &str) -> (i32, String, Str
...
-        .arg(format!(
-            r#"
-            cd $(git rev-parse --show-toplevel)
-            ./scripts/integration_tests.sh -a {} -e {} -A
-            "#,
-            abi, test_name
-        ))
-        .output()
-        .expect("Failed to execute command");
-
```
Comment (juanky201271) [inline rust/zingomobile_utils/src/lib.rs]: You need to put this back, because is to run the tests in mac.

**Review by juanky201271 — CHANGES_REQUESTED** [review]

**Review by juanky201271 — DISMISSED** [review]

**Review by juanky201271 — APPROVED** [review]
This is nice.

**Pull request merged**